### ISSUE TITLE
HyperParameterOptimizer: Add support for logarithmic and reverse logarithmic float/int hyperparams

### ISSUE THREAD

**Issue by iirekm** [allegroai/trains#244]
they both are needed, eg logarithmic for learning rate hyperparams, reverse logarithmic eg for gamma parameter in reinforcement learning, logaritmic int param can be sometimes good eg for example for number of neurons in layer

**Comment by iirekm** [comment]
eg how it's done in SageMager (https://aws.amazon.com/blogs/machine-learning/amazon-sagemaker-automatic-model-tuning-now-supports-random-search-and-hyperparameter-scaling/ )

**Comment by bmartinn** [comment]
Hi @iirekm 
Do you know if `optuna` for example supports it? 
The other options is, we would need to do the linear to log scaling internally and expose only the linear scale to the extarnal implementations ...

**Comment by iirekm** [comment]
Optuna has log hyperparameters (just add log=True to suggest_xxxx), unfortunately doesn't have reverse log, have to be simulated with 1 - log parameter

**Comment by bmartinn** [comment]
Hmm that means we would have to simulate the entire feature so it is available to all optimizer.

I guess we could inherit from [UniformParameterRange](https://github.com/allegroai/trains/blob/45af97b258d9693c787467b050abb5b04f3ec8da/trains/automation/parameters.py#L111) and just log/exp the parent choice.

**Comment by patrickdasilva-clearml** [comment]
Hi @iirekm , we're closing this issue as stale since it's been created too long ago. If you think the issue is still relevant, please make sure the issue persists on a recent version of ClearML (ideally the latest) before re-opening the issue. We'll gladly look at it again.
